[feat] Thicken principle-tdd and principle-clean-code to peer depth (#270)
lugassawan/swe-workbench: skills/principle-clean-code/SKILL.md

| Rule | Guideline |
|------|-----------|
| **Function length** | Prefer under 20 lines. Extract when doing two things. |
| **Naming** | Name reveals intent. No abbreviations except universal ones (ctx, err, id). |
| **Abstraction level** | One level per function. Don't mix SQL strings with business logic. |
| **Comments** | Explain WHY, not WHAT. If code needs WHAT comments, rename or extract. |
| **Error handling** | Handle at the appropriate layer. Don't swallow errors silently. |
| **Argument count** | 0 is ideal; 1 is common; 2 is acceptable; 3+ is suspicious. A boolean flag argument is a hidden second function — split it. |

> | **Command-Query Separation** | A function either changes state (command) or returns a value (query), never both. Mixed functions are hard to compose and reason about. |

Author: **Low** — This table row is two sentences and noticeably wider than all peer rows, breaking the at-a-glance scanning rhythm of the table. The explanation reads well but could be trimmed to a single clause and let the body text (if added later) carry the detail.

```suggestion
| **Command-Query Separation** | A function either changes state or returns a value — not both. |
```

Author: Addressed in f9235f674cbbc1e1913b869de501fe088df140e5: trimmed to "A function either changes state or returns a value — not both." to match single-clause peer rows.

The second caller is not required. The boundary is.

## Boy Scout Rule — scope-bounded
*Leave the code slightly better than you found it — within the diff under review.*
- **Fair game** — rename a poorly named variable, extract a multi-line condition, remove a dead comment — inside files already touched by the PR.

> - **Not fair game** — repo-wide renames or refactors triggered by noticing unrelated code; those belong in a dedicated `principle-refactoring` session.

Author: **Low** — The Boy Scout Rule section introduces a `principle-refactoring` backtick pointer that is structurally identical to the `principle-testing` one removed from `principle-tdd` earlier. Issue #239's AC explicitly states 'no new cross-skill dependencies.' If the auto-load scanner treats backtick-wrapped skill names as dependency signals, this reintroduces the same violation in the clean-code file.

```suggestion
- **Not fair game** — repo-wide renames or refactors triggered by noticing unrelated code; those belong in a dedicated refactoring session.
```

Author: Addressed in b4abada655b5e10666bff368fd9de3ee346feecd: removed the backtick — now reads 'a dedicated refactoring session' in plain prose.
